connectors/sources/network_drive.py

```diff
@@ -116,9 +116,10 @@ async def _remote_validation(self, advanced_rules):
                 validation_message=e.message,
             )
 
-        self.source.create_connection()
+        loop = asyncio.get_running_loop()
+        await loop.run_in_executor(executor=None, func=self.source.create_connection)
```

Comment:
You used `asyncio.to_thread` in different PR, is there any difference?

Reply:
nope, they are the same. to_thread uses run_in_executor inside it 

<img width="599" alt="image" src="https://github.com/elastic/connectors/assets/90465691/6586d897-bfb6-44cf-8a9b-4ebc653a2f55">

Reply:
Can you change this PR to also do `to_thread` for consistency then?